airflow/providers/google/cloud/hooks/dataprep.py

```diff
@@ -0,0 +1,80 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains Google Dataprep hook.
+"""
+
+import requests
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from airflow import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+URL = "https://api.clouddataprep.com/v4/jobGroups/"
+
+
+class GoogleDataprepHook(BaseHook):
+    """
+    Hook for connection with Dataprep API.
+
+    """
+
+    def __init__(self, dataprep_conn_id: str = "dataprep_conn_id") -> None:
+        super().__init__()
+        self.dataprep_conn_id = dataprep_conn_id
+        self._url = URL
+
+    @property
+    def _headers(self):
+        """
+        To limit logic in __init__ method request headers are moved to property.
+        """
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self._token}",
+        }
+        return headers
+
+    @property
+    def _token(self) -> str:
+        """
+        To get connection Dataprep with Airflow you need Dataprep token.
+        It should be added to the Connection in Airflow in JSON format.
+        """
+
+        connection = self.get_connection(self.dataprep_conn_id)
+        try:
+            return connection.extra_dejson["token"]
+        except KeyError:
+            raise AirflowException(
+                "Dataprep token is missing or has invalid format. "
+                "Please make sure that Dataprep token is added to the Airflow Connections."
+            )
```

Comment:
As discusses somewhere in last few days, it's not the best practice to use exception in this way
```suggestion
        token = connection.extra_dejson.get("token")
        if token is None:
            raise AirflowException(
                "Dataprep token is missing or has invalid format. "
                "Please make sure that Dataprep token is added to the Airflow Connections."
            )
        return token
```

Reply:
Done